### ISSUE TITLE
How to set user attribute from domain model?

### ISSUE THREAD

**Issue by ciancu** [oracle/weblogic-deploy-tooling#492]
I have an user model which creates some groups and users:
```
    Security:
        Group:
            USAR_ADMIN:
                Description: The USAR Admin group
            USAR_EF_USER:
                Description: The USAR User group
        User:
            itar_admin:
                Password: welcome1
                GroupMemberOf: [ USAR_ADMIN ]
            itar_user:
                Password: welcome1
                GroupMemberOf: [ USAR_EF_USER ]
```
I will need to set the user email attribute, WLST that can achive this is:
cd('/SecurityConfiguration/usar_domain/Realms/myrealm/AuthenticationProviders/DefaultAuthenticator')
cmo.setUserAttributeValue('itar_admin','displayname','Itar')
cmo.setUserAttributeValue('itar_admin','givenname','Admin')
cmo.setUserAttributeValue('itar_admin','mail','itar_admin@some.com')

Is there a way to achieve the above from the domain model ?

**Comment by ddsharpe** [comment]
Not today.  The user/group pieces were added for offline only.  To update the additional attributes, the tool would need to be enhanced to support online (and the additional fields).

**Comment by ddsharpe** [comment]
I used cmo.setUserAttributeValue() to set "mail".  Then, I used Migration from the WLS console to export the DAT (LDIFT) to get the result below.  The import of the same appears to work as well. The LDIFT format appears to support additional attributes:
dn: uid=derek,ou=people,ou=@realm@,dc=@domain@
objectclass: top
objectclass: person
objectclass: organizationalPerson
objectclass: inetOrgPerson
objectclass: wlsUser
cn: derek
sn: derek
uid: derek
userpassword:: e3NzaGEyNTZ9aG9hbzRFUEtpNVVtd29Ga0NWUGZrTERyMWdwcHZLS2l4eE05dnBKaWpoMVVMS2dO
createTimestamp: 202200032301Z
creatorsName: cn=Admin
mail: first.last@mycompany.com
modifyTimeStamp: 202200032311Z
modifiersName: cn=Admin

Adding support for the additional attributes in offline WDT should be possible.
